lisa/sut_orchestrator/baremetal/build.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+import re
+from typing import List, Type
+
+from smb.SMBConnection import SMBConnection  # type: ignore
+
+from lisa import schema
+from lisa.util import InitializableMixin, subclasses
+from lisa.util.logger import get_logger
+
+from .schema import BuildSchema, FileSchema, SMBBuildSchema
+
+
+class Build(subclasses.BaseClassWithRunbookMixin, InitializableMixin):
+    def __init__(self, runbook: BuildSchema) -> None:
+        super().__init__(runbook=runbook)
+        self._log = get_logger("cluster", self.__class__.__name__)
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return BuildSchema
+
+    def copy(self, source_path: str, files_map: List[FileSchema]) -> None:
+        raise NotImplementedError()
+
+
+class SMBBuild(Build):
+    def __init__(self, runbook: SMBBuildSchema) -> None:
+        super().__init__(runbook)
+        self.smb_runbook: SMBBuildSchema = self.runbook
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "smb"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return SMBBuildSchema
+
+    def copy(self, source_path: str, files_map: List[FileSchema]) -> None:
+        username = self.smb_runbook.username
+        password = self.smb_runbook.password
+        server_name = self.smb_runbook.server_name
+        share_name = self.smb_runbook.share
+
+        conn = SMBConnection(
```

Comment:
Check if the SMBConnection support context, so it can be used by `with`. So the connection will be closed, if there is any exception.

Reply:
updated